### PR TITLE
Replace `wiki-links` flag with `documentation` flag.

### LINKED ISSUE #1331: The `--wiki-links` option has an obsolete name.
Would be nice to change this for the 5.0 release.

Options:
* `--documentation`
* I don't have any other ideas :slightly_smiling_face:.

### PR DESCRIPTION
**Pull request by troessner**
Fixes #1331

### CODE REVIEW

File: features/command_line_interface/options.feature [hunk 1]
```diff
@@ -60,7 +60,7 @@ Feature: Reek can be controlled using command-line options
       Text format options:
               --[no-]color                 Use colors for the output (default: true)
           -V, --[no-]empty-headings        Show headings for smell-free source files (default: false)
-          -U, --[no-]wiki-links            Show link to related wiki page for each smell (default: true)
+          -U, --[no-]documentation            Show link to related wiki page for each smell (default: true)
```
Comment (mvz) [inline features/command_line_interface/options.feature]: -> 'related documentation page'

Comment (troessner) [reply features/command_line_interface/options.feature]: Good point, I also noticed that i missed quite some other internal references to "wiki", lemme fix this.

Comment (troessner) [reply features/command_line_interface/options.feature]: Oh boy, there are a LOT of references to wiki that we had missed 😆

**Comment by troessner** [comment]
Ready for review - again! That was more work than I had expected ;)

File: features/reports/json.feature [hunk 2]
```diff
@@ -24,7 +24,7 @@ Feature: Report smells using simple JSON layout
               "context": "Smelly#x",
               "lines": [ 4 ],
               "message": "has the name 'x'",
-              "wiki_link": "https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md",
+              "documentation_link": "https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Method-Name.md",
```
Comment (mvz) [inline features/reports/json.feature]: :+1:

**Pull request merged**